include/openmc/weight_windows.h

```diff
@@ -0,0 +1,99 @@
+#ifndef OPENMC_WEIGHT_WINDOWS_H
+#define OPENMC_WEIGHT_WINDOWS_H
+
+#include <cstdint>
+#include <unordered_map>
+
+#include <hdf5.h>
+#include <pugixml.hpp>
+
+#include "openmc/constants.h"
+#include "openmc/memory.h"
+#include "openmc/mesh.h"
+#include "openmc/particle.h"
+#include "openmc/vector.h"
+
+namespace openmc {
+
+//==============================================================================
+// Non-member functions
+//==============================================================================
+
+//! Apply weight windows to a particle
+//! \param[in] p  Particle to apply weight windows to
+void apply_weight_windows(Particle& p);
+
+//! Free memory associated with weight windows
+void free_memory_weight_windows();
+
+//==============================================================================
+// Global variables
+//==============================================================================
+
+class WeightWindows;
+
+namespace variance_reduction {
+
+extern std::unordered_map<int32_t, int32_t> ww_map;
+extern vector<unique_ptr<WeightWindows>> weight_windows;
+
+} // namespace variance_reduction
+
+//==============================================================================
+//! Individual weight window information
+//==============================================================================
+
+struct WeightWindow {
+  double lower_weight {-1}; // -1 indicates not valid state
+  double upper_weight {1};
+  double survival_weight {0.5};
+  double weight_cutoff {WEIGHT_CUTOFF};
+  int max_split {1};
+
+  //! Whether the weight window is in a valid state
+  operator bool() const { return lower_weight >= 0.0; }
```

Comment:
I think it would be more clear to make this into a method called `.valid()` or `.is_valid()`. The semantic meaning of the boolean value of something called a `WeightWindow` is not immediately apparent to a newcomer to this code.

Reply:
I can see this either way. I think this version is a little more elegant and less verbose to use, but I agree with your point on it being potentially confusing to newcomers. I'll go ahead and make it a `valid` method as you've suggested.